### PR TITLE
Add more user management

### LINKED ISSUE #111: Admin to ban user

### PR DESCRIPTION
**Pull request by nginyc**
Fixes #111 

Before testing, since Rafiki's v0.1.0 Docker images have not been published to Docker Hub, will need to run `source .env.sh`, then `bash scripts/build_images.sh` for every participating machine.

- Admins can ban users
- Admins can list all users
- Admins cannot ban other admins (only superadmins can do this)
- Admins cannot ban themselves
- Admins cannot create a user with an invalid type
- Add documentation on new user APIs

### CODE REVIEW

File: rafiki/admin/app.py [hunk 1]
```diff
@@ -20,27 +20,62 @@ def index():
...
+    params = get_request_params()
+
+    # Expect csv file as bytes
+    csv_file_bytes = request.files['csv_file_bytes'].read()
+    params['csv_file_bytes'] = csv_file_bytes
+
+    with admin:
+        return jsonify(admin.create_users_with_csv(**params))
+
+@app.route('/users', methods=['POST'])
```
Comment (nudles) [inline rafiki/admin/app.py]: It is not common to create users using a csv file. Typically, we add users one by one. But we need to get all users for the front-end.
Have you compared users vs user for the operations against a single user? Both are fine, but we need to be consistent with other APIs. E.g., model vs models, trial vs trials.

Comment (nginyc) [reply rafiki/admin/app.py]: Noted. i will be removing this API

**Comment by nginyc** [comment]
Changed from deleting user to banning user to avoid complication of having to consider cascade-deleting all associated train jobs / trials / inference jobs / models under that user.

**Comment by wild-flame** [comment]
Can add to documentation 

for sudo users need to run 

```sudo -E bash scripts/build_images.sh``` after run `source env.sh`

 according to https://stackoverflow.com/questions/8633461/how-to-keep-environment-variables-when-using-sudo

to preserve the environment variable

**Comment by nginyc** [comment]
Thanks @wild-flame for the feedback. I've added that to the documentation.

**Pull request merged**